components/raftstore/src/store/peer.rs

```diff
@@ -4686,7 +4685,7 @@ where
     ) -> bool {
         let pending_snapshot = self.is_handling_snapshot() || self.has_pending_snapshot();
         // shouldn't transfer leader to witness peer or non-witness waiting data
-        if self.is_witness() || self.wait_data
+        if self.is_witness() || is_learner(&self.peer) || self.wait_data
```

Comment:
it's checked by `ready_to_transfer_leader`

Reply:
If the leader hasn't applied, the check in `ready_to_transfer_leader` is based on stale configuration, leading in incorrect result.